src/libexpr/nixexpr.hh

```diff
@@ -278,6 +333,8 @@ struct ExprLambda : Expr
     Symbol arg;
     Formals * formals;
     Expr * body;
+    DocComment docComment;
```

Comment:
I almost fell for this again. We need to link lambdas and the comments that may occur on the attribute instead of the lambda. We could do that differently, with an extra map, but the implementation of that would be quite ugly that way.